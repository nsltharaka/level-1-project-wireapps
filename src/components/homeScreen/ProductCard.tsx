import { Image, StyleSheet, Text, View } from "react-native";
import React from "react";
import type { Product } from "@/types/product";

type Props = {
  item: Product;
};

export default function ProductCard({ item }: Props) {
  return (
    <View style={styles.container}>
      <Image source={{ uri: item.mainImage }} style={styles.image} />
      <Text>{item.name}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "48%",
    height: 250,
    borderRadius: 16,
    padding: 10,
    backgroundColor: "#fff",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
  },
  image: {
    width: "100%",
    height: 200,
    objectFit: "contain",
  },
});
