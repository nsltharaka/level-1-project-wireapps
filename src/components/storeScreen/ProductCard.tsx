import type { Product } from "@/types/product";
import React from "react";
import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";

type Props = {
  item: Product;
  onPressHandleFunc: (itemId: string) => void;
};

export default function ProductCard({ item, onPressHandleFunc }: Props) {
  return (
    <TouchableOpacity
      onPress={() => onPressHandleFunc(item.id)}
      style={styles.container}
    >
      <Image source={{ uri: item.mainImage }} style={styles.image} />
      <Text style={styles.itemName} numberOfLines={2}>
        {item.name}
      </Text>
      <Text>{item.colour}</Text>
      <Text>$ {item.price}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "48%",
    borderRadius: 16,
    padding: 10,
    backgroundColor: "#fff",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
  },
  image: {
    width: "100%",
    height: 200,
    objectFit: "contain",
  },
  itemName: {
    fontSize: 16,
    fontWeight: "bold",
    height: 40,
    marginBottom: 5,
  },
});
