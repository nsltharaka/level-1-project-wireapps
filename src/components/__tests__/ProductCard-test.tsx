import { render } from "@testing-library/react-native";
import ProductCard from "../storeScreen/ProductCard";

import FavoritesContextProvider from "@/contexts/favorites/FavoritesContext";
import sampleData from "@/data/sample.json";
import { type PropsWithChildren } from "react";

describe("<ProductCard />", () => {
  const Wrapper = ({ children }: PropsWithChildren) => {
    return <FavoritesContextProvider>{children}</FavoritesContextProvider>;
  };
  test("Product card renders correctly with all optional configs", () => {
    const tree = render(
      <ProductCard item={sampleData[0]}>
        <ProductCard.Color />
        <ProductCard.FavoriteButton />
        <ProductCard.Price />
      </ProductCard>,
      { wrapper: Wrapper },
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
