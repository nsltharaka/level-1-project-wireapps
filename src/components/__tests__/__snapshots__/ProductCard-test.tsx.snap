// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> Product card renders correctly with all optional configs 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 16,
      "elevation": 3,
      "flex": 1,
      "gap": 5,
      "opacity": 1,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/7e386191b2ee40b290886a05d3e10e24_nike-air-relentless-a.jpg",
      }
    }
    style={
      {
        "height": 200,
        "objectFit": "contain",
        "width": "100%",
      }
    }
  />
  <Text
    numberOfLines={2}
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
        "height": 40,
        "marginBottom": 5,
      }
    }
  >
    Nike Air Relentless 4 Mens Running Shoes
  </Text>
  <Text>
    blue
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "padding": 7,
        "position": "absolute",
        "right": 5,
        "top": 5,
      }
    }
  >
    <Text />
  </View>
  <Text>
    $ 
    45.00
  </Text>
</View>
`;

exports[`<ProductCard /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 16,
      "elevation": 3,
      "flex": 1,
      "gap": 5,
      "opacity": 1,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/7e386191b2ee40b290886a05d3e10e24_nike-air-relentless-a.jpg",
      }
    }
    style={
      {
        "height": 200,
        "objectFit": "contain",
        "width": "100%",
      }
    }
  />
  <Text
    numberOfLines={2}
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
        "height": 40,
        "marginBottom": 5,
      }
    }
  >
    Nike Air Relentless 4 Mens Running Shoes
  </Text>
  <Text>
    blue
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "padding": 7,
        "position": "absolute",
        "right": 5,
        "top": 5,
      }
    }
  >
    <Text />
  </View>
  <Text>
    $ 
    45.00
  </Text>
</View>
`;
